src/Rules.Framework/Extensions/GenericSearchArgsExtensions.cs

```diff
@@ -0,0 +1,32 @@
+namespace Rules.Framework.Extensions
+{
+    using System;
+    using System.Linq;
+    using Rules.Framework.Generics;
+
+    internal static class GenericSearchArgsExtensions
+    {
+        public static SearchArgs<TContentType, TConditionType> ToSearchArgs<TContentType, TConditionType>(
+            this SearchArgs<GenericContentType, GenericConditionType> genericSearchArgs)
+        {
+            if (!typeof(TContentType).IsEnum)
+            {
+                throw new ArgumentException("Only TContentType of type enum are currently supported.");
+            }
+
+            var contentType = (TContentType)Enum.Parse(typeof(TContentType), genericSearchArgs.ContentType.DisplayName);
+
+            var searchArgs = new SearchArgs<TContentType, TConditionType>(contentType, genericSearchArgs.DateBegin, genericSearchArgs.DateEnd)
+            {
+                Conditions = genericSearchArgs.Conditions.Select(condition => new Condition<TConditionType>
+                {
+                    Value = condition.Value,
+                    Type = (TConditionType)Enum.Parse(typeof(TConditionType), condition.Type.DisplayName)
+                }),
```

Comment:
Perhaps a `ToList()` here to avoid projection at each `Conditions` enumeration? What do you think?

Reply:
Agree, done.